examples/test_interpose.c

```diff
@@ -62,8 +63,8 @@ int main (int argc, char* argv[])
   // prime the file
   for(i=0; i < 1; i++) {
     id++;
-    sprintf(dir,    "%s/checkpoint_set_%d", prefix, id);
-    sprintf(file,   "%s/rank_%d.ckpt", dir, rank);
+    scnprintf(dir, sizeof(dir), "%s/checkpoint_set_%d", prefix, id);
+    scnprintf(file, sizeof(file), "%s/rank_%d.ckpt", dir, rank);
```

Comment:
I would probably use `snprintf` rather than the home-grown `scnprintf` then use `snprintf`'s return value to check if the buffer was actually too small and abort rather than continue with the truncated value. Ie:

```
int len = snprintf(dir, sizeof(dir), "%s/checkpoint_set_%d", prefix, id);
if((size_t)len >= sizeof(dir) scr_abort(blah);
```

otherwise this seems to only patch part of the issue.

Reply:
Fixed in latest push